rpc/block.go

```diff
@@ -51,7 +51,7 @@ func WithBlockTag(tag string) BlockID {
 
 // BlockWithTxHashes gets block information given the block id.
 func (provider *Provider) BlockWithTxHashes(ctx context.Context, blockID BlockID) (interface{}, error) {
```

Comment:
What do you think about returning a concrete type here? We could use an adapter function, and return an error if we don't get the expected type. Right now it can return anything

Reply:
Actually, nvm, I don't think I'm correct here. Currently the function should either return BlockTxHashes or a PendingBlocktxHashes

Reply:
I left it this way to be similar to the other method we have `BlockWithTxs`. If we change this one I believe it's a good idea to change the other as well